packages/sdk-react/src/components/receivables/Filters/Filters.tsx

```diff
@@ -18,118 +16,152 @@ import {
   Select,
   MenuItem,
 } from '@mui/material';
+import { DatePicker } from '@mui/x-date-pickers';
 
 import {
   FILTER_TYPE_SEARCH,
   FILTER_TYPE_STATUS,
   FILTER_TYPE_CUSTOMER,
+  FILTER_TYPE_DUE_DATE_LTE,
 } from '../consts';
 
 type Props = {
-  onChangeFilter: (field: keyof FilterTypes, value: FilterValue) => void;
+  onChangeFilter: ReceivablesFilterHandler;
+  filters: Array<keyof FilterTypes>;
 };
 
-export const Filters = ({ onChangeFilter }: Props) => {
+export const Filters = ({ onChangeFilter, filters }: Props) => {
   const { i18n } = useLingui();
   const counterpartQuery = useCounterpartList();
   const { root } = useRootElements();
 
   return (
     <Grid container spacing={2}>
-      <Grid item sm={6} md={4}>
-        <SearchField
-          label={t(i18n)`Search`}
-          onChange={(search) => {
-            onChangeFilter(FILTER_TYPE_SEARCH, search);
-          }}
-        />
-      </Grid>
-      <Grid item sm={3} md={2}>
-        <MuiFormControl variant="outlined" fullWidth>
-          <MuiInputLabel id="status">{t(i18n)`Status`}</MuiInputLabel>
-          <Select
-            labelId="status"
-            label={t(i18n)`Status`}
-            defaultValue="all"
-            MenuProps={{ container: root }}
-            onChange={(selected) => {
-              onChangeFilter(
-                FILTER_TYPE_STATUS,
-                selected && selected.target.value
-              );
+      {filters.includes('search') && (
+        <Grid item sm={6} md={4}>
+          <SearchField
+            label={t(i18n)`Search`}
+            onChange={(search) => {
+              onChangeFilter(FILTER_TYPE_SEARCH, search);
+            }}
+          />
+        </Grid>
+      )}
+      {filters.includes('status') && (
+        <Grid item sm={3} md={2}>
+          <MuiFormControl variant="outlined" fullWidth>
+            <MuiInputLabel id="status">{t(i18n)`Status`}</MuiInputLabel>
+            <Select
+              labelId="status"
+              label={t(i18n)`Status`}
+              defaultValue="all"
+              MenuProps={{ container: root }}
+              onChange={(selected) => {
+                onChangeFilter(
+                  FILTER_TYPE_STATUS,
+                  selected.target.value === 'all'
+                    ? null
+                    : (selected.target.value as ReceivablesStatusEnum)
+                );
+              }}
+            >
+              {[
+                { label: t(i18n)`All statuses`, value: 'all' },
+                ...Object.values(ReceivablesStatusEnum).map((status) => ({
+                  label: (() => {
```

Comment:
What do you think of moving this logic into a function constant? To not re-create an object on each update.
We could create a function constant by `key: ReceivablesStatusEnum` and a value `t(i18n)`

Reply:
I don't see a problem with creating this object and iterating each time. It's not a 1000-line list.

Reply:
But it's anyway worse than creating a function 1 time and not re-creating it on each render. If we will create a new function in each component it will affect performance. I see no point in not moving it into a separate function to avoid any memory costs when it's not needed

Reply:
You're talking about premature optimization. Memoization is much more expensive than iterating an object that is created and quickly deleted. Also, all this is wonderfully optimized by all known JS engines.

It's worth it to optimize the components that really slow things down. Also, React Compiler will address the need for optimization on its own in the future. And this is not some distant future.